src/include/function/gds/gds_frontier.h

```diff
@@ -79,66 +161,57 @@ class GDSFrontier {
  * All functions supported in this base interface are thread-safe.
  */
 class Frontiers {
-    friend class GDSTask;
+    friend class FrontierTask;
 
 public:
-    explicit Frontiers(GDSFrontier* curFrontier, GDSFrontier* nextFrontier,
-        uint64_t initialActiveNodes)
-        : curFrontier{curFrontier}, nextFrontier{nextFrontier} {
+    explicit Frontiers(std::shared_ptr<GDSFrontier> curFrontier,
```

Comment:
If we already have `Frontier`, then `Frontiers` is not a good name. 

`XXXs` is normally reserved for `vector<XXX>`. If we need a class to represent multiple XXX, the convention is to name as `XXXCollection` in your case. `FrontierPair` is more accurate.